### PR TITLE
[toolchain] fix warnings/errors in Keil compiler

### PR DESCRIPTION
**Pull request by LuDuda**
This PR fixes all warnings and errors which were reported by ARM Keil compiler.

Basically there were three bigger problems, FYI:

1. Keil demands that all unions/structures inside packed class/structure have to be packed.

Examples: 

`otIp6Address `is packed structure, but `mFields `union inside this structure is not.

Non-packed `otIp6Prefix `structure is a member of DHCPv6 packed classes.

Packed `Mac::Frame` class inherits from non-packed `otRadioFrame` - here however since `Mac::Frame` doesn't have any data members, i removed packed qualifier from it, but i can pack `otRadioFrame` if that is the preferred way.

2. Differences between __packed uint8_t and non-packed uint8_t.

In the newly added spinel decoder module, there is `ReadUint8(uint8_t &aUint8)` method.

Unfortunately Keils throws non-supressable error on code like that:

```
otIp6Prefix prefix;  // Note that this structure is packed now.
mDecoder.ReadUint8(prefix.mLength);  // mLength has __packed uint8_t type, however function requires non-packed uint8_t.
```

I have created a local non-packed variable as a workaround.

3. Keil throws warnings on unreachable statement.

I simply suppressed this warning in configuration file, since it looks very suspicious :) Alternatively i can add `OT_UNREACHABLE_CODE` macro to all cases as shown below (but i would not recommend it).

Examples:

```
assert(false);
break;   <-- Warning, unreachable statement.
```

or

```
ExitNow(error = ..)
break;   <-- Warning, unrechable statement.
```

-------------------------------------------------------------------------

If you have any suggestions on how to solve above issues in better way, please let me know.

I see also that i can remove __packed qualifier from `otIp6Prefix` by removing __packed qualifier from `PrefixAgent ` and `IdentityAssociation` classes, if this is the preferred way (this also solves issue 2 without touching existing code).

### CODE REVIEW

File: src/ncp/ncp_base_mtd.cpp [hunk 1]
```diff
@@ -1136,8 +1134,7 @@ otError NcpBase::InsertPropertyHandler_THREAD_OFF_MESH_ROUTES(void)
 
     VerifyOrExit(mAllowLocalNetworkDataChange == true, error = OT_ERROR_INVALID_STATE);
 
-    SuccessOrExit(error = mDecoder.ReadIp6Address(routeConfig.mPrefix.mPrefix));
-    SuccessOrExit(error = mDecoder.ReadUint8(routeConfig.mPrefix.mLength));
+    SuccessOrExit(error = mDecoder.ReadIp6Prefix(routeConfig.mPrefix));
```
Comment (abtink) [inline src/ncp/ncp_base_mtd.cpp]: How about if we fix the packed vs. unpacked issue by using a local `uint8_t prefixLength` variable that we pass to `ReadUint8()` instead and then assign the read value to `routeConfig.mPrefix.mLength`.

Even though adding a `ReadIp6Prefix()` makes the parsing code smaller, the intention behind `SpinelDecoder/Encoder` is for them to provide `Read/Write<Type>` methods mirroring the [primitive Spinel types](https://github.com/openthread/openthread/blob/master/doc/spinel-protocol-src/spinel-data-packing.md#primitive-types) and IP6 Prefix is not really a basic/primitive Spinel supported type. 

Thoughts?

Comment (LuDuda) [reply src/ncp/ncp_base_mtd.cpp]: Sure, that was my initial idea, but i decided to implement ReadIp6Prefix method. I will make needed changes, thanks!

Comment (abtink) [reply src/ncp/ncp_base_mtd.cpp]: Thanks.

File: src/core/net/dhcp6_server.cpp [hunk 2]
```diff
@@ -428,7 +428,7 @@ otError Dhcp6Server::SendReply(otIp6Address &aDst, uint8_t *aTransactionId, Clie
     SuccessOrExit(error = AppendRapidCommit(*message));
 
     memset(&messageInfo, 0, sizeof(messageInfo));
-    memcpy(&messageInfo.GetPeerAddr().mFields.m8, &aDst, sizeof(otIp6Address));
+    memcpy(messageInfo.GetPeerAddr().mFields.m8, &aDst, sizeof(otIp6Address));
```
Comment (abtink) [inline src/core/net/dhcp6_server.cpp]: Nice catch!

**Pull request merged**